hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/scope/HandleHederaNativeOperations.java

```diff
@@ -90,31 +94,29 @@ public void setNonce(final long contractNumber, final long nonce) {
      */
     @Override
     public @NonNull ResponseCodeEnum createHollowAccount(@NonNull final Bytes evmAddress) {
-        throw new AssertionError("Not implemented");
+        final var synthTxn = TransactionBody.newBuilder()
+                .cryptoCreateAccount(synthHollowAccountCreation(evmAddress))
+                .build();
+        final var childRecordBuilder = context.dispatchChildTransaction(synthTxn, CryptoCreateRecordBuilder.class);
+        // TODO - switch OK to SUCCESS once some status-setting responsibilities are clarified
+        if (childRecordBuilder.status() != OK) {
+            throw new AssertionError("Not implemented");
+        }
+        return OK;
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public void finalizeHollowAccountAsContract(@NonNull final Bytes evmAddress) {
-        throw new AssertionError("Not implemented");
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public void collectFee(final long fromEntityNumber, final long amount) {
-        throw new AssertionError("Not implemented");
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public void refundFee(final long fromEntityNumber, final long amount) {
-        throw new AssertionError("Not implemented");
+        requireNonNull(evmAddress);
+        final var accountStore = context.readableStore(ReadableAccountStore.class);
+        final var hollowAccountId = requireNonNull(accountStore.getAccountIDByAlias(evmAddress));
+        final var tokenServiceApi = context.serviceApi(TokenServiceApi.class);
+        tokenServiceApi.finalizeHollowAccountAsContract(hollowAccountId, INITIAL_CONTRACT_NONCE);
+        // For temporary backward-compatibility with mono-service, consume an entity id
```

Comment:
So we used to consume  an entity id when finalizing?  How strange.

Reply:
Reasons. 🤷